openshift/insights-operator
# OCPBUGS-16017: update DataGather CR status in case of job failure

This is only for TechPreview and it updates the `DataGather` custom resource status in case of failure. The `datagather_conditions.go` file has been renamed to `datagather_status.go` and some functions were moved there (to be able to reuse them easily). 

Steps to verify:

1. Get TechPreview cluster
2. Switch to `openshift-insights` namespace
3. Wait for some `periodic-gathering` to happen and delete the Pod when it's running
4. Check the corresponding DataGather custom resource with the same name

## Categories


- [ ] Data Enhancement

## Sample Archive


- no new data

## Documentation



## Unit Tests


- `pkg/controller/status/datagather_status_test.go`

## Privacy


Yes. There are no sensitive data in the newly collected information.

## Changelog


## Breaking Changes


No

## References


https://issues.redhat.com/browse/OCPBUGS-16017
https://access.redhat.com/solutions/???
